src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    private FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public static FruitTransaction of(String operationCode, String fruit, int quantity) {
```

Comment:
You can implement this as usual constructor approach and get rid of using `static`.
There will be no problem with constructor it will be overloaded because it has different type of parameters.